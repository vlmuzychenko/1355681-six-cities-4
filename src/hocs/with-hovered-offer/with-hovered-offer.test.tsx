import React from "react";
import renderer from "react-test-renderer";
import withHoveredOffer from "./with-hovered-offer";

const offerMock = {
  id: 1,
  title: `Beautiful & luxurious apartment at great location`,
  price: 120,
  previewImage: `img/apartment-01.jpg`,
  images: [`img/room.jpg`, `img/apartment-01.jpg`, `img/apartment-02.jpg`, `img/apartment-03.jpg`, `img/studio-01.jpg`, `img/apartment-01.jpg`],
  description: `An independent House, strategically located between Rembrand Square and National Opera, but where the bustle of the city comes to rest in this alley flowery and colorful.`,
  rating: 4,
  isPremium: true,
  type: `apartment`,
  bedrooms: 4,
  maxAdults: 4,
  goods: [`Wi-Fi`, `Washing machine`, `Towels`, `Heating`, `Coffee machine`, `Baby seat`, `Kitche`, `Dishwasher`, `Cabel TV`, `Fridge`],
  host: {
    name: `Angelina`,
    super: true,
    avatarUrl: `img/avatar-angelina.jpg`,
  },
  coords: [52.3909553943508, 4.85309666406198],
  city: {
    name: `Amsterdam`,
    coords: [52.377956, 4.897070],
  },
};

const MockComponent = () => {
  return (
    <div></div>
  );
};

const MockComponentWrapped = withHoveredOffer(MockComponent);

it(`withHoveredOffer is rendered correctly`, () => {
  const tree = renderer.create((
    <MockComponentWrapped
      onOfferHover={() => {}}
      hoveredOffer={offerMock}
    />
  ), {
    createNodeMock() {
      return {};
    }
  }).toJSON();

  expect(tree).toMatchSnapshot();
});
