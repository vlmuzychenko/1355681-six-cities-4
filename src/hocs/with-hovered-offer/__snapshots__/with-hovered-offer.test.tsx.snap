// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`withHoveredOffer is rendered correctly 1`] = `<div />`;
